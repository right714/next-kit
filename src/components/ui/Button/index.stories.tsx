import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";

import Button from "./index";

export default {
  title: "Components/Button",
  component: Button,
  argTypes: {
    children: { control: "text" },
    backgroundColor: { control: "color" },
  },
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = args => <Button {...args}>{args.children}</Button>;

export const Default = Template.bind({});
Default.args = {
  children: "ボタン",
};
