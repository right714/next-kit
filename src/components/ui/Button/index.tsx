import { FC } from "react";
import styled from "./style.module.scss";
import classnames from "classnames";

const cx = classnames.bind(styled);

type ButtonProps = {
  children?: React.ReactNode;
};

export default (props => {
  const { children } = props;

  return (
    <button type="button" className={cx(styled.button, "text-red-500")}>
      {children}
    </button>
  );
}) as FC<ButtonProps>;
