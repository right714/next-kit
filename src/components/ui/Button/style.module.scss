@use "@/styles/helper" as *;

.button {
  border: 1px solid gray;

  @include mq(pc) {
    font-size: 1.2em;
  }
}
