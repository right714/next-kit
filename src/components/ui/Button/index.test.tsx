import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import Button from "./";

test("Button", () => {
  const { container } = render(<Button>ボタン</Button>);

  expect(container).toMatchSnapshot();
});
