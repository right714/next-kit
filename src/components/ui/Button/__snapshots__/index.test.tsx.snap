// Vitest Snapshot v1

exports[`Button 1`] = `
<div>
  <button
    class="_button_3bd438 text-red-500"
    type="button"
  >
    ボタン
  </button>
</div>
`;
